cognoml/analysis.py

```diff
@@ -0,0 +1,114 @@
+import collections
+import os
+import warnings
+
+import pandas as pd
+import numpy as np
+import sklearn
+from sklearn.model_selection import train_test_split
+
+from cognoml import utils
+from cognoml.classifiers.logistic_regression import grid_search
+from cognoml.figshare import download_files
+
+# expression_path = os.path.join('download', 'mutation-matrix.tsv.bz2')
+data_directory = "download"
+
+def read_data(version=None):
+    """
+    Read data.
+    """
+    v_dir = download_files(directory=data_directory, artile_id=3487685, version=version)
```

Comment:
> For mutable defaults only?

No for all defaults. Their value is evaluated upon definition. This leads to an odd behavior for mutable defaults where they can be modified with each function call. Immutables don't have this issue, but are still evaluated upon definition.